File: engine/model/src/main/scala/net/server/channel/packet/mts/PlaceItemForSalePacket.scala
package net.server.channel.packet.mts

class PlaceItemForSalePacket(private var _available: Boolean, private var _operation: Byte, private var _itemId: Int, private var _quantity: Short, private var _price: Int, private var _slot: Short) extends BaseMTSPacket(_available = _available, _operation = _operation) {
  def itemId: Int = _itemId

  def quantity: Short = _quantity

  def price: Int = _price

  def slot: Short = _slot
}

File: engine/engine-model/src/main/java/net/server/channel/packet/mts/PlaceItemForSalePacket.java
package net.server.channel.packet.mts;

public class PlaceItemForSalePacket extends BaseMTSPacket {
   private final Integer itemId;

   private final Short quantity;

   private final Integer price;

   private final Short slot;

   public PlaceItemForSalePacket(Boolean available, Byte operation, Integer itemId, Short quantity, Integer price, Short slot) {
      super(available, operation);
      this.itemId = itemId;
      this.quantity = quantity;
      this.price = price;
      this.slot = slot;
   }

   public Integer itemId() {
      return itemId;
   }

   public Short quantity() {
      return quantity;
   }

   public Integer price() {
      return price;
   }

   public Short slot() {
      return slot;
   }
}
